### PR TITLE
e2e: Loop more to defeat service randomization

### LINKED ISSUE #18958: e2e flake: Services should work after restarting kube-proxy
http://kubekins.dls.corp.google.com:8080/view/Critical%20Builds/job/kubernetes-e2e-gke-ci/10042/

expected to retrieve pods
     [service1-48igt service1-57dir service1-i1ojq],
only retrieved
    [service1-i1ojq service1-i1ojq service1-48igt service1-i1ojq service1-48igt service1-48igt service1-i1ojq service1-48igt service1-48igt] 

@thockin

### PR DESCRIPTION
**Pull request by thockin**
Fixes #18958 (hopefully)

### CODE REVIEW

File: test/e2e/service.go [hunk 1]
```diff
@@ -1368,8 +1370,6 @@ func verifyServeHostnameServiceUp(c *client.Client, ns, host string, expectedPod
 		passed := false
 		pods := []string{}
 		// Retry cmdFunc for upto 5 minutes.
```
Comment (ixdy) [inline test/e2e/service.go]: the comment here suggests the functions would be retried for up to 5 minutes, though the code actually only retries for 1 minute.

in any case, shouldn't we increase the timeout here? or am I misunderstanding this code?

Comment (thockin) [inline test/e2e/service.go]: I can fix the comment - I think a minute is reasonable.

**Comment by ixdy** [comment]
I still don't think this will work.

Already the code retries several times for a minute, but this isn't enough time. Looping more in the command probably isn't going to help, especially with no additional timeouts there.
```
Dec 21 00:17:29.170: INFO: Creating new exec pod
STEP: verifying service has 3 reachable backends
Dec 21 00:17:31.180: INFO: Executing cmd set -e; for i in $(seq 1 9); do wget -q -T 1 -O - http://10.247.255.190:80 || true; echo; done on host 104.197.155.143:22
Dec 21 00:18:02.404: INFO: Waiting for expected pods for 10.247.255.190: [service1-48igt service1-57dir service1-i1ojq], got: [service1-48igt service1-i1ojq]
Dec 21 00:18:07.404: INFO: Executing cmd set -e; for i in $(seq 1 9); do wget -q -T 1 -O - http://10.247.255.190:80 || true; echo; done on host 104.197.155.143:22
Dec 21 00:18:15.599: INFO: Waiting for expected pods for 10.247.255.190: [service1-48igt service1-57dir service1-i1ojq], got: [service1-48igt service1-i1ojq]
Dec 21 00:18:20.599: INFO: Executing cmd set -e; for i in $(seq 1 9); do wget -q -T 1 -O - http://10.247.255.190:80 || true; echo; done on host 104.197.155.143:22
Dec 21 00:18:41.798: INFO: Waiting for expected pods for 10.247.255.190: [service1-48igt service1-57dir service1-i1ojq], got: [service1-48igt service1-i1ojq]
STEP: deleting pod execpod in namespace e2e-tests-services-iw9l6
Dec 21 00:18:46.807: INFO: Unexpected error occurred: service verification failed for:
 10.247.255.190, expected to retrieve pods [service1-48igt service1-57dir service1-i1ojq], only retrieved [service1-i1ojq service1-i1ojq service1-48igt service1-i1ojq service1-48igt service1-48igt service1-i1ojq service1-48igt service1-48igt]
```

**Comment by thockin** [comment]
This isn't about waiting for a timeout, it's about defeating the randomization in the kernel to ensure each endpoint gets hit at least once.  3x is not enough, even 50x is not deterministic, but there's just no way to make this deterministic - it is probabilistic by implementation.

If 50x isn't enough, I'll try 100x.

**Comment by ixdy** [comment]
OK, thanks for the explanation - I think I understand now. LGTM.

**Comment by ihmccreery** [comment]
@thockin This is causing this test to take hours to fail, causing entire builds to time out.  I'd like to propose a revert.

**Comment by ihmccreery** [comment]
xref https://github.com/kubernetes/kubernetes/issues/20561

**Pull request merged**